src/Pagination.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import BootstrapMixin from './BootstrapMixin';
+
+const Pagination = React.createClass({
+  mixins: [BootstrapMixin],
+
+  propTypes: {
+    activePage: React.PropTypes.number,
+    items: React.PropTypes.number,
+    maxButtons: React.PropTypes.number,
+    ellipsis: React.PropTypes.bool,
+    first: React.PropTypes.bool,
+    last: React.PropTypes.bool,
+    prev: React.PropTypes.bool,
+    next: React.PropTypes.bool,
+    onSelect: React.PropTypes.func
+  },
+
+  getDefaultProps() {
+    return {
+      activePage: 1,
+      items: 1,
+      maxButtons: null,
+      first: false,
+      last: false,
+      prev: false,
+      next: false,
+      ellipsis: true,
+      bsClass: 'pagination'
+    };
+  },
+
+  _onSelect(event) {
+    if(this.props.onSelect && event.currentTarget.className.search('disabled') === -1){
+      this.props.onSelect(parseInt(event.currentTarget.dataset.pagenumber));
+    }
+  },
+
+  noClick(event){
+    event.preventDefault();
+  },
+
+  renderPageButtons() {
+    let pageButtons = [];
+
+    let startPage, endPage, hasMoreStart, hasMoreEnd;
+
+    if(this.props.maxButtons){
+      hasMoreStart = this.props.activePage - 1 > 1;
+      startPage = hasMoreStart ? this.props.activePage - 1 : 1;
+      hasMoreEnd = startPage + this.props.maxButtons <= this.props.items;
+
+      if(!hasMoreEnd){
+        endPage = this.props.items;
+        startPage = this.props.items - this.props.maxButtons + 1;
+      } else {
+        endPage = startPage + this.props.maxButtons - 1;
+      }
+    } else {
+      startPage = 1;
+      endPage = this.props.items;
+    }
+
+    for(let pagenumber = startPage; pagenumber <= endPage; pagenumber++){
+      pageButtons.push(
+        <li
+          key={pagenumber}
+          className={pagenumber === this.props.activePage ? 'active' : null}
+          data-pagenumber={pagenumber}
+          onClick={this._onSelect}>
+          <a href='#' onClick={this.noClick}>{pagenumber}</a>
+        </li>
+      );
+    }
+
+    if(this.props.maxButtons && hasMoreEnd && this.props.ellipsis){
+      pageButtons.push(
+        <li key='showMoreEnd' className='disabled'><a href='#' onClick={this.noClick}>...</a></li>
+      );
+    }
+
+    return pageButtons;
+  },
+
+  renderPrev() {
+    return this.props.prev ? (
+      <li
+        data-pagenumber={this.props.activePage - 1}
```

Comment:
What value does this data attribute bring? We are not using Bootstrap's JavaScript so the data tags are unnecessary in this library.